### ISSUE TITLE
Azure templates automatically expanded may break existing CI

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule.Rules.Azure#796]
**Description of the issue**

PSRule for Azure **1.4.0** automatically expands template parameter files.
If incomplete templates that were not evaluated previously already exist in the repository this may break CI processes.

**Expected behaviour**

This feature should be opt-in by setting the `AZURE_PARAMETER_FILE_EXPANSION` configuration option.

i.e.

```yaml
# YAML: Set the AZURE_PARAMETER_FILE_EXPANSION configuration option to enable expansion
configuration:
  AZURE_PARAMETER_FILE_EXPANSION: true
```

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **1.4.0**

Captured output from `$PSVersionTable`:

```text
Name                           Value
----                           -----
PSVersion                      7.1.3
PSEdition                      Core
GitCommitId                    7.1.3
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```
